include/cxxopts.hpp

```diff
@@ -1538,6 +1538,22 @@ CXXOPTS_DIAGNOSTIC_POP
     return CXXOPTS_RTTI_CAST<const values::standard_value<T>&>(*m_value).get();
   }
 
+#ifdef CXXOPTS_HAS_OPTIONAL
+  template <typename T>
+  std::optional<T>
+  optional() const
+  {
+    try
+    {
+      return as<T>();
```

Comment:
Our thought is that cxx still ought to throw for type mismatch (asking for int as a string, day), and return `std::nullopt` if the name lookup fails (undefined value, missing CLI parameter).

Reply:
That seems like user error though, since you should know what types you created your options with. In which case it should still throw.

Reply:
Yes, agreed.  Will ammend.